diff -Naur r5u870_k2.6.27.org/usbcam/usbcam_fops.c r5u870_k2.6.27/usbcam/usbcam_fops.c
--- r5u870_k2.6.27.org/usbcam/usbcam_fops.c	2009-12-30 10:01:49.000000000 +0000
+++ r5u870_k2.6.27/usbcam/usbcam_fops.c	2009-12-30 10:02:28.000000000 +0000
@@ -1195,7 +1195,7 @@
 	.mmap		= usbcam_v4l_mmap,
 	/*.ioctl		= video_ioctl2,*/
 	.ioctl		= usbcam_v4l_ioctl,
-#ifdef CONFIG_COMPAT
+#if defined(CONFIG_COMPAT) && LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 29)
 	.compat_ioctl	= v4l_compat_ioctl32,
 #endif
 };
